zbus/src/connection/socket/unixexec.rs

```diff
@@ -0,0 +1,118 @@
+use std::io;
+use std::os::fd::BorrowedFd;
+
+#[cfg(feature = "tokio")]
+use tokio::io::{AsyncReadExt, ReadBuf};
+
+use crate::process::{Child, ChildStdin, ChildStdout};
+
+use super::{ReadHalf, RecvmsgResult, Socket, Split, WriteHalf};
+
+#[derive(Debug)]
+pub struct UnixExecStream {
```

Comment:
* I'm wondering if this is really unix-specific? AFAICT, it's not but only currently used for a unix-specific address type. If so, I'd give it a more generic name but for now only enable the mod/use for unix target. E.g `CommandStream`?
* Would be good to split addition of this socket type in a separate commit (especially if we make it more generic) that comes before its use in `connection::Builder`.
* I don't think this should be public API. Users just need a `Connection` given a unixexec address, and we can handle all that internally.
* Some docs would be good in any case. :)